src/plugins/media/schema.js

```diff
@@ -1,44 +1,43 @@
-// Copyright 2018 OpenStax Poland
+// Copyright 2019 OpenStax Poland
 // Licensed under the MIT license. See LICENSE file in the project root for
 // full license text.
 
 function normalizeMedia(change, error) {
-    const { code: violation, key, node } = error
+    const { code: violation } = error
 
     switch (violation) {
-    case 'node_data_invalid':
-        if (key === 'alt') {
-            change.setNodeByKey(node.key, {
-                data: { alt: 'Cannot be empty' },
-            })
-        }
-        break
-
     /* istanbul ignore next */
     default:
         console.warn("Unhandled media violation", violation)
         break
     }
 }
 
-export default {
-    blocks: {
-        // TODO: do we actually want to keep multiple versions?
-        media: {
-            nodes: [
-                { match: [{ type: 'image' }], min: 1 },
-            ],
-            data: {
-                alt: Boolean,
+export default function schema({ marks }) {
+    return {
+        blocks: {
+            // TODO: do we actually want to keep multiple versions?
+            media: {
+                nodes: [
+                    { match: { type: 'image' }, min: 1 },
+                    { match: { type: 'media_alt' }, min: 0, max: 1 },
+                    { match: { type: 'media_text' }, min: 0, max: 1 },
+                ],
+                normalize: normalizeMedia,
             },
-            normalize: normalizeMedia,
-        },
-        // TODO: if we do, add target attribute.
-        image: {
-            isVoid: true,
-            data: {
-                src: Boolean,
+            // TODO: if we do, add target attribute.
+            image: {
+                isVoid: true,
+                data: {
+                    src: Boolean,
+                },
+            },
+            media_alt: {
+                marks: marks.map(type => ({ type })),
+            },
```

Comment:
Since alt-text is ultimately going to be stored in the `alt` attribute it can't contain marks.